src/Streamly/Internal/Data/Stream/IsStream/Top.hs

```diff
@@ -601,8 +660,8 @@ unionBy eq s1 s2 =
 --
 -- Space: O(1)
 --
--- /Unimplemented/
-{-# INLINE mergeUnionBy #-}
-mergeUnionBy :: -- (IsStream t, Monad m) =>
+-- /Pre-release/
+{-# INLINE unionBySorted #-}
+unionBySorted :: (IsStream t, MonadAsync m, Semigroup (t m a)) =>
     (a -> a -> Ordering) -> t m a -> t m a -> t m a
-mergeUnionBy _eq _s1 _s2 = undefined
+unionBySorted eq s1 s2 = s1 <> Stream.removeDupsRight eq s1 s2
```

Comment:
1. We should not use the stream `s1` twice, that's a functional behavior issue.
2. This is inefficient because we go through the stream s1 twice, performance issue.

Instead we can go through both the streams in tandem only once and union them.